pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -177,7 +177,17 @@ func mergeService(dst, src *model.Service, srcRegistry serviceregistry.Instance)
 	// prefer the k8s VIP where possible
 	clusterID := srcRegistry.Cluster()
 	if srcRegistry.Provider() == provider.Kubernetes || len(dst.ClusterLocal.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
-		dst.ClusterLocal.ClusterVIPs.SetAddressesFor(clusterID, []string{src.Address})
+		dst.ClusterLocal.ClusterVIPs.SetAddressesFor(clusterID, src.ClusterLocal.ClusterVIPs.GetAddressesFor(clusterID))
+	}
+	if srcRegistry.Provider() == provider.Kubernetes {
+		// Copy over Kubernetes Multi-Cluster Services (MCS) ClusterSet Host and IPs.
+		if len(dst.ClusterSetLocal.Hostname) == 0 {
```

Comment:
How come we need hostname here but not for L180 ClusterLocal?

Reply:
same doubt